qcodes/instrument_drivers/QuTech/M2j.py

```diff
@@ -0,0 +1,83 @@
+from qcodes.instrument.base import Instrument
+from qcodes.utils.validators import Numbers
+import numpy as np
+
+try:
+    from spirack import M2j_module
+except ImportError:
+    raise ImportError(('The M2j_module class could not be found. '
+                       'Try installing it using pip install spirack'))
+
+
+class M2j(Instrument):
+
+    def __init__(self, name, spi_rack, module, **kwargs):
+        """
+        Qcodes driver for the M2j RF amplifier SPI-rack module.
+
+        Args:
+            name (str): name of the instrument.
+
+            spi_rack (SPI_rack): instance of the SPI_rack class as defined in
+                the spirack package. This class manages communication with the
+                individual modules.
+
+            module (int): module number as set on the hardware.
+
+        The gain and RF_level can only be set to the device, but not read from the device.
+
+        Parameters:
+            gain: Amplifier gain in dB, range 32 to 55 dB
+            RF_level: Measured RF power after amplification (not calibrated)
+        """
+        super().__init__(name, **kwargs)
+
+        self.m2j = M2j_module.M2j_module(spi_rack, module)
+
+        self.add_parameter('gain',
+                           label='gain',
+                           set_cmd=self._set_gain,
+                           unit='dB',
+                           vals=Numbers())
+
+        self.add_parameter('RF_level',
+                           label='RF level',
+                           get_cmd=self._meas_power,
+                           unit='dBm')
+
+        self.add_function('clear_rf_clip',
+                          call_cmd=self.m2j.clear_rf_clip)
+        self.add_function('is_rf_clipped',
+                          call_cmd=self.m2j.rf_clipped)
+
+    def _set_gain(self, dB):
+        if dB < 32:
+            print(
+                "notice, lower limit for the gain is 32dBm. The limit will be the set value.")
```

Comment:
what do you mean with "the limit will be the set value"? So far as I read it, it means that the limit is going to be changed, which is not true. I would rephrase this to `f"An attempt is made to set the gain to {dB} which is below 32dBm. This is not possible. The gain will be set to 32 dBm."`

Also, a better design is to add the limits to the `add_parameter` function via `vals` argument like `vals=Numbers(min_value=32, max_value=55)` ([see for example this driver for an example](https://github.com/QCoDeS/Qcodes/blob/fb574ef1cf2b7a76cdab5d6591cb4f7a5eced376/qcodes/instrument_drivers/stanford_research/SR86x.py#L619))